sea-orm-macros/src/derives/model.rs

```diff
@@ -1,95 +1,74 @@
 use crate::{
     attributes::derive_attr,
-    util::{escape_rust_keyword, field_not_ignored, trim_starting_raw_identifier},
+    util::{escape_rust_keyword, trim_starting_raw_identifier},
 };
+use darling::{ast, util, FromDeriveInput, FromField};
 use heck::CamelCase;
 use proc_macro2::TokenStream;
-use quote::{format_ident, quote, quote_spanned};
-use std::iter::FromIterator;
-use syn::{punctuated::Punctuated, token::Comma, Ident, Lit, Meta};
+use quote::{format_ident, quote};
+use syn::Ident;
 
 enum Error {
-    InputNotStruct,
+    Parsing(darling::Error),
     Syn(syn::Error),
 }
 
+#[derive(Debug, FromField)]
+#[darling(attributes(sea_orm), allow_unknown_fields = true)]
+struct DeriveField {
+    ident: Option<Ident>,
+    #[darling(default)]
+    ignore: bool,
+    #[darling(default)]
+    enum_name: Option<String>,
+    #[darling(skip)]
+    column_ident: Option<Ident>,
+}
+
+#[derive(Debug, FromDeriveInput)]
+#[darling(supports(struct_named))]
```

Comment:
this will make sure that we only support `struct` and not `enum`.

so I replaced  `Error::InputNotStruct` with `Parsing(darling::Error)`

and changed handling  in `pub fn expand_derive_model` respectively.